File: challenges/aoc/template.cpp
#include <fstream>
#include <iostream>
#include <string>
#include <chrono>

using namespace std;

int main() {
    ifstream file("sample.txt");
    // ifstream file("input.txt");

    // for testing performance
    chrono::steady_clock::time_point start = chrono::steady_clock::now();

    string content;
    string line;
    while (std::getline(file, line)) {
        content += line + "\n";
    }

    file.close();

    cout << "File content:\n" << content;

    // for testing performance
    chrono::steady_clock::time_point end = chrono::steady_clock::now();
    chrono::duration<double> elapsedSeconds = end - start;
    cout << "Execution Time: " << elapsedSeconds.count() << " seconds" << endl;

    return 0;
}

File: challenges/aoc/template.py
import time

with open("sample.txt", "r") as file:
    # with open("input.txt", 'r') as file:
    lines = file.readlines()

# only for testing performance
start_time = time.time()

content = ""
for line in file:
    content += line + "\n"

print("File content:\n", content)

# only for testing performance
end_time = time.time()
execution_time = end_time - start_time
print("Execution Time:", execution_time, "seconds")
